helix-gateway/src/main/java/org/apache/helix/gateway/service/GatewayServiceManager.java

```diff
@@ -91,20 +91,26 @@ public void onGatewayServiceEvent(GatewayServiceEvent event) {
     }
   }
 
-  private GatewayCurrentStateCache getCache(String clusterName) {
-    return _currentStateCacheMap.computeIfAbsent(clusterName, k -> new GatewayCurrentStateCache(clusterName));
-  }
-
   public void resetTargetStateCache(String clusterName, String instanceName) {
     getCache(clusterName).resetTargetStateCache(instanceName);
   }
 
+  /**
+   * Overwrite the current state cache with the new current state map, and return the diff of the change.
+   * @param clusterName
+   * @param newCurrentStateMap
+   * @return
+   */
   public  Map<String, Map<String, Map<String, String>>> updateCacheWithNewCurrentStateAndGetDiff(String clusterName,
       Map<String, Map<String, Map<String, String>>> newCurrentStateMap) {
    return  getCache(clusterName).updateCacheWithNewCurrentStateAndGetDiff(newCurrentStateMap);
```

Comment:
The naming is not that good. Could be getCacheInstance? What if the get is null? Then would that getOrCreateCacheInstance?

Reply:
Yes if get is null, it create instance. I will update name.